integration-tests/public/events/Cargo.toml

```diff
@@ -28,3 +28,10 @@ std = [
 ]
 ink-as-dependency = []
 e2e-tests = []
+
+[profile.test]
+# Need this for linkme crate to work for the event metadata collection unit test.
+# See https://github.com/dtolnay/linkme/issues/61#issuecomment-1503653702
+# NOTE: `cargo-contract` also sets this flag when generating metadata for the same reason.
+# See https://github.com/use-ink/cargo-contract/pull/1200
+lto = "thin"
```

Comment:
NOTE: This is just being re-introduced, it was present for this test for previous ink! versions (e.g. [see here](https://github.com/use-ink/ink/blob/84d1158de43e0f7d8f0eb87843e931d33643f897/integration-tests/public/events/Cargo.toml#L32-L35)) and was erroneously remove by https://github.com/use-ink/ink/commit/03a8eeeab690d291d0ac9bd75c84bb015c8b6532#diff-a00395296d1808215704bc7cca551820d8232fab091345f72ca021aaccb239cf

I added a few more comments here to clarify it's purpose